app/controllers/announcements_controller.rb

```diff
@@ -47,6 +47,10 @@ def set_announcement
       @announcement = Announcement.find(params[:id])
     end
 
+    def determine_layout
+      "without_navbar" unless @system_setting.display_navbar
```

Comment:
Minor: i think rails generates a `display_navbar?` method with the `?` suffix. Might a tad more idiomatic/expressive to use that one?